src/target/nixos.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2015-2017 Intecture Developers. See the COPYRIGHT file at the
+// top-level directory of this distribution and at
+// https://intecture.io/COPYRIGHT.
+//
+// Licensed under the Mozilla Public License 2.0 <LICENSE or
+// https://www.tldrlegal.com/l/mpl-2.0>. This file may not be copied,
+// modified, or distributed except according to those terms.
+
+use command::{CommandResult, CommandTarget};
+use directory::DirectoryTarget;
+use error::{Error, Result};
+use file::{FileTarget, FileOwner};
+use host::Host;
+use package::PackageTarget;
+use package::providers::Providers;
+use serde_json::Value;
+use service::ServiceTarget;
+use std::{env, process, str};
+use std::path::Path;
+use super::{default_base as default, linux_base as linux};
+use host::telemetry::{Cpu, Os, Telemetry, TelemetryTarget};
+
+pub struct NixOsTarget;
+
+//
+// Command
+//
+
+impl CommandTarget for NixOsTarget {
+    #[allow(unused_variables)]
+    fn exec(host: &mut Host, cmd: &str) -> Result<CommandResult> {
```

Comment:
Yes that was why. However, I did this before you removed the `BinResolver` so let me give it another shot now. Perhaps it's now working fine using the default implementation.